cocotb/triggers.py

```diff
@@ -729,3 +729,74 @@ def _wait(self):
         for _ in range(self.num_cycles):
             yield trigger
         raise ReturnValue(self)
+
+
+class StableCondition(Waitable):
+    """
+    Wait until a given condition is True in ReadOnly state.
+
+    The trigger will first do ``yield ReadOnly()`` and then do the check.
+
+    Args:
+        condition (callable): Output of condition called without any parameters
+            is checked to be equivalent of logic True for condition to be met.
+        event (Trigger): event after which to check the condition; defaults to
+            nexttimestep.
+        extra_event (bool): whether to wait for an extra event after condition
+            is met; defaults to ``False``.
+
+    Example:
+
+    .. code-block:: python
+
+        # Wait for ack
+        yield StableCondition(
+            lambda: bus.cycle.integer and bus.ack.integer,
+            event=RisingEdge(dut.clk)
+        )
+
+    .. versionadded: 1.3
+    """
+    def __init__(self, condition, **kwargs):
+        event = kwargs.pop('event', NextTimeStep())
+        extra_event = kwargs.pop('extra_event', False)
+        reject_remaining_kwargs('StableCondition.__init__', kwargs)
+        self.condition = condition
+        self.event = event
+        self.extra_event = extra_event
+
+    @decorators.coroutine
+    def _wait(self):
+        while True:
+            yield ReadOnly()
+            if self.condition():
+                break
+            yield self.event
+        if self.extra_event:
+            yield self.event
+
+
+class StableValue(StableCondition):
+    """
+    Wait until a signal has a given value in ReadOnly state.
+
+    The trigger will first do ``yield ReadOnly()`` and then do the comparison.
+
+    Args:
+        signal (SimHandle): the signal to monitor.
+        event (Trigger): event after which next is done; default is waiting for
+            a change of the given signal; defaults to ``Edge(signal)``.
```

Comment:
Perhaps add - "Note that this is equivalent to passing RisingEdge when value == 1, and FallingEdge when value == 0

Reply:
For ``StableValue`` the default for ``event`` changes from ``NextTimeStep`` to ``Edge``; that's why I could not pass kwargs on as is.

Reply:
Will change StableValue docstring to:
```python
class StableValue(StableCondition):
    """
    Wait until a signal has a given value in ReadOnly state.

    The trigger will first do ``yield ReadOnly()`` and then do the comparison.

    Args:
        signal (SimHandle): the signal to monitor.
        value (int, 0 or 1): value to wait for.
        **kwargs: keyword arguments passed on to :class:`StableCondition`
            The default for ``event`` is changed to ``Edge(signal)``. This is
            equivalent to ``RisingEdge(signal)`` when waiting for 1 and
            ``FallingEdge(signal)`` when waiting for 0.

    .. versionadded: 1.3
    """
```